API/YAML design: Custom URL for MDM routes (#22547)
fleetdm/fleet: docs/REST API/rest-api.md

      "enable_end_user_authentication": false,
      "macos_setup_assistant": "path/to/config.json",
      "enable_release_device_manually": true
    },

>     "client_url": "https://instance.fleet.com"

Reviewer: Just to be sure, this is at `features.client_url` right?

Author: Hey @jacobshandling, I think no. We want this to be in the `mdm` object. So it would be `mdm.client_url`.

Reviewer: Right, realized that but for some reason my comment didn't post. Thanks for the clarification:
<img width="817" alt="Screenshot 2024-10-08 at 1 59 12 PM" src="https://github.com/user-attachments/assets/a6bf2192-6739-4064-b723-606e9d44a3cb">

| macos_setup         | object  | See [`mdm.macos_setup`](#mdm-macos-setup). |
| macos_settings         | object  | See [`mdm.macos_settings`](#mdm-macos-settings). |
| windows_settings         | object  | See [`mdm.windows_settings`](#mdm-windows-settings). |

> | client_url         | string  | Update this URL if you're self-hosting Fleet and you want your hosts to talk to a different URL for MDM features (default: base URL of the Fleet instance provided during setup). |

Reviewer 2: ```suggestion
| client_url         | string  | Update this URL if you're self-hosting Fleet and you want your hosts to talk to a different URL for MDM features (default: base URL of the Fleet instance provided during setup). |
```
@noahtalerman does this mean it uses the base URL of the Fleet instance if empty, or that it gets filled in with the Fleet URL upon setup or migration & has to be manually updated if that base URL ever changes?

Reviewer 2: IMO would be a better user experience to allow this to be empty and then default to the base URL if empty, rather than requiring the user to update in 2 places if they don't need a custom client URL and their base URL changes, which would look more like:
```suggestion
| client_url         | string  | Update this URL if you're self-hosting Fleet and you want your hosts to talk to a different URL for MDM features. (If not configured, `server_url` will be used instead.) |
```

Author: > better user experience to allow this to be empty and then default to the base URL if empty, rather than requiring the user to update in 2 places if they don't need a custom client URL and their base URL changes

@rachaelshaw agreed!

Maybe it's clearer w/o the "provided during setup?"

(If not configured, hosts will use the base URL of the Fleet instance.)
